tests/Yaapii.Atoms.Tests/IO/WriterAsOutputStreamTest.cs

```diff
@@ -0,0 +1,128 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System;
+using System.IO;
+using System.Text;
+using Xunit;
+using Yaapii.Atoms.Bytes;
+using Yaapii.Atoms.Enumerable;
+using Yaapii.Atoms.Text;
+
+namespace Yaapii.Atoms.IO.Tests
+{
+    public sealed class WriterAsOutputStreamTest
+    {
+        [Fact]
+        public void WritesContentToFile()
+        {
+            var dir = "artifacts/WriterAsOutputStreamTest"; var inputFile = "large-text.txt";
+            var inputPath = Path.GetFullPath(Path.Combine(dir, inputFile));
+            var outputFile = "text-copy.txt";
+            var outputPath = Path.GetFullPath(Path.Combine(dir, outputFile));
+
+            Directory.CreateDirectory(dir);
+            if (File.Exists(inputPath)) File.Delete(inputPath);
+            if (File.Exists(outputPath)) File.Delete(outputPath);
+
+            //Create large file
+            new LengthOf(
+                new InputOf(
+                    new TeeInputStream(
+                        new MemoryStream(
+                            new BytesOf(
+                                new Text.Joined(", ",
+                                    new HeadOf<string>(
+                                        new Endless<string>("Hello World"),
+                                        1000
+                                    )
+                                )
+                            ).AsBytes()
+                        ),
+                        new OutputTo(
+                            new Uri(inputPath)
+                        ).Stream()
+                    )
+                )
+            ).Value();
+
+            //Read from large file and write to output file (make a copy)
+            var filestream = new FileStream(outputPath, FileMode.OpenOrCreate, FileAccess.Write, FileShare.Write);
+
+            long left =
+            new LengthOf(
+                new InputOf(
+                    new TeeInputStream(
+                        new InputStreamOf(
+                            new Uri(Path.GetFullPath(inputPath))
+                        ),
+                        new WriterAsOutputStream(
+                            new StreamWriter(filestream)
+                        )
+                    )
+                )
+
+            ).Value();
+
+            long right =
+                new LengthOf(
+                    new InputOf(
+                        new Uri(Path.GetFullPath(outputPath))
+                    )
+                ).Value();
+
+            Assert.True(left == right, "input and output are not the same size");
```

Comment:
Put the objects left and right directly into the Assert:
`Assert.True(new LengthOf(...).Value() == new LenghtOf(...).Value(), ...)`